### PR TITLE
Revert "Revert "Implement stat for SB < 16""

### PR DESCRIPTION
**Pull request by madhurajayaraman**
b/302730696

Reverts youtube/cobalt#2553

### CODE REVIEW

**Review by andrewsavage1 — APPROVED** [review]
This LGTM but please get someone working on the Starboard transition to take a closer look, especially for the android changes

File: starboard/android/shared/test_filters.py [hunk 1]
```diff
@@ -60,6 +60,10 @@
         'SbDirectoryGetNextTest.SunnyDayStaticContent',
         'SbDirectoryOpenTest.SunnyDayStaticContent',
         'SbFileGetPathInfoTest.WorksOnStaticContentDirectories',
+        'PosixDirectoryCanOpenTest.SunnyDayStaticContent',
```
[the remark this answers is not in the document]
Comment (madhurajayaraman) [reply starboard/android/shared/test_filters.py:63]: These don't pass when they're not filtered, but I can file a bug for investigating all of the static content failures.

File: starboard/CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,10 @@ since the version previous to it.
 
 ## Version 16
 
+### Added standard POSIX socket getaddrinfo/freeaddrinfo APIs.
```
Comment (y4vor) [inline starboard/CHANGELOG.md:12]: This is not `stat` related, why is it part of this PR?

Comment (madhurajayaraman) [reply starboard/CHANGELOG.md:12]: If you look at the file in the PR in total (not just the later commits) it shows the correct history -- only the stat changelog message is added.

File: starboard/shared/modular/starboard_layer_posix_stat_abi_wrappers.cc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "starboard/shared/modular/starboard_layer_posix_stat_abi_wrappers.h"
+
+int __abi_wrap_stat(const char* path, struct musl_stat* musl_info) {
+  struct stat stat_info;  // The type from platform toolchain.
```
[the remark this answers is not in the document]
Comment (madhurajayaraman) [reply starboard/shared/modular/starboard_layer_posix_stat_abi_wrappers.cc:18]: What kind of error should I throw?

**Pull request merged**